python-client/giskard/models/base/__init__.py

```diff
@@ -244,30 +261,40 @@ def predict(self, dataset: Dataset) -> ModelPredictionResults:
               The `all_predictions` field will contain the predicted probabilities for all class labels for each example in the input dataset.
         """
         timer = Timer()
-        df = self.prepare_dataframe(dataset)
 
-        raw_prediction = self.predict_df(df)
+        # Read cache
+        cached_predictions = self.model_cache.read_from_cache(dataset.df[GISKARD_HASH_COLUMN])
+        missing = pd.isna(cached_predictions)
+        if len(missing.shape) > 1:
+            missing = np.any(missing, axis=1)
```

Comment:
It’s better if the cache returns a dataframe with a single column or a `pd.Series`, and your `isna()` would be sufficient. Otherwise, `cached_predictions.isna().any(axis=1)` will directly give you what you want.